operator/pkg/httphelper/client.go

```diff
@@ -68,11 +67,10 @@ type CassMetadataEndpoints struct {
 }
 
 func BuildPodHostFromPod(pod *corev1.Pod) string {
-	return GetPodHost(
-		pod.Name,
-		pod.Labels[api.ClusterLabel],
-		pod.Labels[api.DatacenterLabel],
-		pod.Namespace)
+	// This function previously returned the dns hostname which includes the StatefulSet's headless service,
+	// which is the datacenter service. There are times though that we want to make a mgmt api call to the pod
+	// before the dns hostnames are available. It is therefore more reliable to simply use the PodIP.
+	return pod.Status.PodIP
```

Comment:
if this returns the empty string, what will happen?

Reply:
I am not sure under what conditions this would return an empty string. Do you want me to add a check and return an error if it is the empty string?

Reply:
I tried checking this on Pending / Terminating pods, and it looks like there are some windows where there won't be a PodP.

Adding an error return value changes the signature and makes this a bigger fix, but that might be the best way to handle this.

Reply:
I updated the function to return an error when there is no IP.